### ISSUE TITLE
[1.2.0] Auto-login function

### ISSUE THREAD

**Issue by jackouille-ch** [Cacti/cacti#1820]
Is it possible to generate a Cacti Session then die within the opened window?

In fact I would like to have a funthe possibility to login in with an an URL such as https:/cacti.ch/cacti/auth_login.php?user=cacti&passwd=passwd
And simply right after Cacti login will be performed then the script will end without displaying anything else.
I use Cacti Graphs on a 3rd party GUI and need to get the connection to Cacti with different credentials per user.

With Cacti  0.8x I was performing the following changes on each release so that it works for me:
Changed /usr/share/cacti/include/csrf/csrf-magic.php  and put $GLOBALS['csrf']['defer'] = true;
And in /usr/share/cacti/auth_login.php at line 199 (before switch ($user["login_opts"])) I add the following:
if (get_request_var_post("MyBreak Code")==1) {
				die();
			}

How can I do the same with 1.2.0 ?

**Comment by netniV** [comment]
No, there is no functionality for that.  However, there is a plugin to export images from the system to a third party one.

For security, you should never pass the username and passed in a get request as that is very easily captured and shows up in browser history.

**Comment by jackouille-ch** [comment]
Yes but export is running only each 5 minutes and I cannot export thousands of graphs, it will take too long.

Regarding security, accounts used have very restricted rights (only viewing their devices) so it's not an issue for me.

**Comment by netniV** [comment]
There may be a way that would not require patching every time ... using a plugin ... I have an idea that would help a few people out in that regard.

**Comment by jackouille-ch** [comment]
Yeah that would be wondeful to have an embedded solution!!!

By the way where can I download all current plugins? Maybe it would be good to include them with the release. From my side I copy plugins from 0.8.8x but not sure that they are all compatible with 1.x

**Comment by netniV** [comment]
OK, so it's a 0.0.1 release but I think this could work for you... it requires some work on your side to generate the urls which will work for a day.

https://github.com/netniv/plugin_tokenauth/

**Comment by jackouille-ch** [comment]
Great, many thanks for your quick job. I will work on it next week and will keep you informed.

**Comment by jackouille-ch** [comment]
I put in production and that works fine but I got caveats.
Maybe some enhancements:
1. Update each hour the token because after 24 hours, token expires despite account is continuously viewing graphs.
2. Possibility to restrict logs with foloowing content: "CMDPHP plugin_tokenauth: No user session found" - option to disable this messages in Configuration / Settings or but maybe not feasible will be group alerts (keep the same alert but indicate the number of occurences)
3. Add an option in Cacti to autoclose itself (after token has been generated) with an option autoclose=1 for example. It will give the following: https://cacti.mydomain.ch/cacti/?tokenauth_id=$id&tokenauth_token=$signature&autoclose=1

**Comment by netniV** [comment]
I would recommend raising any issues for that plugin on it's own repo to be tracked properly.
